sdk/state/update.go

```diff
@@ -91,6 +92,9 @@ func (c *Channel) PaymentTxs(p *Payment) (close, decl *txnbuild.Transaction, err
 }
 
 func (c *Channel) ConfirmPayment(p *Payment) (payment *Payment, fullySigned bool, err error) {
+	if p.IterationNumber != c.IterationNumber() {
+		return nil, fullySigned, errors.New("invalid payment iteration number")
```

Comment:
Can you include the got and want values in this error message? It'll help with debugging I'm guessing.

Reply:
https://github.com/stellar/experimental-payment-channels/pull/77/commits/9b23f3e2e318b766708891ed4bc0711c2c850493